cloudformation-states.yaml

```diff
@@ -0,0 +1,110 @@
+AWSTemplateFormatVersion: "2010-09-09"
+Description: Set up an executable pipeline state machine
+
+Parameters:
+  Environment:
+    Type: String
+    Default: staging
+    AllowedValues:
+      - staging
+      - production
+    Description: The environment for which the cluster need to be created.
+  ExperimentId:
+    Default: '6379'
+    Type: String
+    Description: The ID of the experiment the system is created for.
+
+Resources:
+  PipelineStateMachine:
+    Type: "AWS::StepFunctions::StateMachine"
+    Properties:
+      DefinitionString:
+        !Sub
+          - |-
+            {
+              "Comment": "N/A",
+              "StartAt": "DeleteCompletedJobs",
+              "States": {
+                "DeleteCompletedJobs": {
+                  "Type": "Task",
+                  "Comment": "Deletes all the preivous server jobs that are already completed.",
+                  "Resource": "arn:aws:states:::eks:call",
+                  "Parameters": {
+                    "ClusterName": "biomage-${Environment}",
+                    "CertificateAuthority.$": "$.certificateAuthority",
```

Comment:
`certificateAuthority` is an input parameter that must be added when the pipeline is run. This is because there is no easy way to get the CA from the CloudFormation stack outputs for the cluster. Hopefully we can resolve this problem from the API in the next ticket so we don't need to do this.